cli/src/semdep/parsers/poetry.py

```diff
@@ -94,7 +102,15 @@
 # category = "main"
 # optional = false
 # python-versions = ">=3.6"
-poetry_dep = mark_line(string("[[package]]\n") >> key_value_list.map(lambda x: dict(x)))
+poetry_dep = mark_line(
+    string("[[package]]\n")
+    >> key_value_list.map(
+        lambda x: {
+            key_val[0]: ValueLineWrapper(line_number, key_val[1])
```

Comment:
Main thing to get reviewed tbh @mmcqd

Reply:
making mark_line return a ValueLineWrapper will still require a ValueLineWrapper constructor here because of how we are parsing the key_value_list so not much benefit IMO

Reply:
the change here will be 


```
poetry_dep = mark_line(
    string("[[package]]\n")
    >> key_value_list.map(
        lambda x: {
            obj.value[0]: ValueLineWrapper(line_number, obj.value[1])
            for obj in x
        }
    )
)
```

after changing mark_line, which looks pretty much the same